### ISSUE TITLE
Html entities not converting

### ISSUE THREAD

**Issue by 0bon** [solidjs/solid#1088]
### Describe the bug

I create a componant that passes some text with some html entity like `&amp;`. The `&amp;` does not converted into & - it remains as is.

### Your Example Website or App

n/a

### Steps to Reproduce the Bug or Issue

```
// file_a.ts  ===========================
<Button text='&amp; text' />

// button.ts  ===========================
interface Props {
    text: string;
}

export default (props: Props) => (
    <div>
        <a href="#">{props.text}</a>
    </div>
);
```

### Expected behavior

n/a

### Platform

solidjs 1.4.4

**Comment by edemaine** [comment]
Indeed, this seems to be intended behavior. In JavaScript strings, you write `'& text'`. In HTML, you write `&amp; text` for the same content. JSX needs to bridge this gap. [The spec says](https://facebook.github.io/jsx/#sec-jsx-string-characters):

> Historically, string characters within [JSXAttributeValue](https://facebook.github.io/jsx/#prod-JSXAttributeValue) and [JSXText](https://facebook.github.io/jsx/#prod-JSXText) are extended to allow the presence of [HTML character references](https://facebook.github.io/jsx/#sec-HTMLCharacterReference) to make copy-pasting between HTML and JSX easier, at the cost of not supporting \ [EscapeSequence](https://tc39.es/ecma262/#prod-annexB-EscapeSequence) of ECMAScript's [StringLiteral](https://tc39.es/ecma262/#prod-StringLiteral). We may revisit this decision in the future.

This would suggest that your example should work. But Solid's choices seem to be a little more nuanced. When using actual HTML elements, the attributes get character references expanded, as in the JSX spec, which makes sense for copy/pasting HTML:

```jsx
<span title="&amp; text">&amp; text</span>
// translates to template(`<span title="& text">&amp; text</span>`)
// both get rendered as ampersands
```

When passing in attributes to a component though, the character references don't get expanded, so you end up with raw `&amp;`. This is somewhat reasonable because you can't be copy/pasting HTML at this point... but I could see an argument either way.

I noticed an actual bug though:

```jsx
<span title="&quot; text">&quot; text</span>
// translates to invalid HTML:
template(`<span title="" text">&quot; text</span>`)
```

**Comment by ryansolid** [comment]
Yeah I think I missed static attributes (hoisted into the template). Thanks for reporting. I will see what can be done. Babel parser gives back a value and a raw value, and sometimes those need additional massaging.

**Comment by ryansolid** [comment]
Yeah I guess this one is by design and my attribute handling is inconsistent.. static strings being hoisted should be escaped. So:
```js
<span title="&amp; text">&amp; text</span>
// translates to
template(`<span title="&amp; text">&amp; text</span>`)
```
However that still generates them both as ampersands. I will fix this as part #1091. But closing this one as by design.
